sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java

```diff
@@ -49,6 +52,123 @@ public PagedIterable(PagedFlux<T> pagedFlux) {
      */
     @SuppressWarnings("deprecation")
     public <S> PagedIterable<S> mapPage(Function<T, S> mapper) {
-        return new PagedIterable<>(pagedFlux.mapPage(mapper));
+        //return new PagedIterable<>(pagedFlux.mapPage(mapper));
+        return new PagedIterableImpl<T, S>(this, mapper);
+    }
+
+    private static final class IteratorImpl<T, S> implements Iterator<S> {
+
+        private final Iterator<T> source;
+        private final Function<T, S> mapper;
+
+        private IteratorImpl(Iterator<T> source, Function<T, S> mapper) {
+            this.source = source;
+            this.mapper = mapper;
+        }
+
+        @Override
+        public boolean hasNext() {
+            return source.hasNext();
+        }
+
+        @Override
+        public S next() {
+            return mapper.apply(source.next());
+        }
+
+        @Override
+        public void remove() {
+            source.remove();
+        }
+    }
+
+    private static final class IterableImpl<T, S> implements Iterable<S> {
+
+        private final Iterable<T> source;
+        private final Function<T, S> mapper;
+
+        private IterableImpl(Iterable<T> source, Function<T, S> mapper) {
+            this.source = source;
+            this.mapper = mapper;
+        }
+
+        @Override
+        public Iterator<S> iterator() {
+            return new IteratorImpl<T, S>(source.iterator(), mapper);
+        }
+    }
+
+    private static final class PagedIterableImpl<T, S> extends PagedIterable<S> {
+
+        private final PagedIterable<T> source;
+        private final Function<T, S> mapper;
+        private final Function<PagedResponse<T>, PagedResponse<S>> pageMapper;
+
+        private PagedIterableImpl(PagedIterable<T> sourcePageIterable, Function<T, S> mapper) {
+            super(new PagedFlux<S>(Mono::empty));
```

Comment:
this is a pain

Reply:
We could move the other paging iterables into `implementation` to re-use the code and add any additional hooks to allow mapping of pages to be performed

Reply:
It there is no plan to deprecate this API (as with `PagedFlux.mapPage`), and if the implementation is done correctly, I believe this solution would be superior than the `PagedFlux.mapPage` approach.

Let's take time for review and discussion. Meanwhile I will switch our usage to code like this, so no urgency.